packages/api/src/models/medical/cohort.ts

```diff
@@ -0,0 +1,109 @@
+import { Cohort, CohortAssignment, MonitoringSettings } from "@metriport/core/domain/cohort";
+import { DataTypes, Sequelize } from "sequelize";
+import { BaseModel, BaseModelNoId, ModelSetup } from "../_default";
+
+export class CohortModel extends BaseModel<CohortModel> implements Cohort {
+  static NAME = "cohort";
+
+  declare cxId: string;
+  declare name: string;
+  declare monitoring?: MonitoringSettings;
+  declare otherSettings?: Record<string, unknown>;
+
+  static override attributes() {
+    return {
+      ...super.attributes(),
+      cxId: {
+        type: DataTypes.UUID,
+        allowNull: false,
+      },
+      name: {
+        type: DataTypes.STRING,
+        allowNull: false,
+      },
+      monitoring: {
+        type: DataTypes.JSONB,
+      },
+      otherSettings: {
+        type: DataTypes.JSONB,
+      },
+    };
+  }
+
+  static setup: ModelSetup = (sequelize: Sequelize) => {
+    CohortModel.init(
+      {
+        ...BaseModel.attributes(),
+        cxId: {
+          type: DataTypes.UUID,
+        },
+        name: {
+          type: DataTypes.STRING,
+          allowNull: false,
+        },
+        monitoring: {
+          type: DataTypes.JSONB,
+        },
+        otherSettings: {
+          type: DataTypes.JSONB,
+        },
+      },
+      {
+        ...BaseModel.modelOptions(sequelize),
+        tableName: this.NAME,
+      }
+    );
+  };
+}
+
+export class CohortAssignmentModel
+  extends BaseModelNoId<CohortAssignmentModel>
```

Comment:
re: BaseModelNoId - usually worth having an id on every single table, even join tables, regardless of whether there's another valid unique identifier for a row. This is one of those things where the cost is relatively low, and the value of convention is high imo. Curious what @leite08 thinks.

e.g. in some crazy world maybe we want to have the row that points to a PatientCohort row for some reason - to do that cleanly we'd want to have a row uuid to point to.